tests/NuGet.Jobs.Catalog2Registration.Tests/Hives/HiveMergerFacts.FullEnumeration.cs

```diff
@@ -0,0 +1,319 @@
+                {
+                    var minVersion = allUpdated.Min(x => x.Version);
+                    for (int i = 0; i < indexInfo.Items.Count - 1; i++)
+                    {
+                        if (indexInfo.Items[i].Upper < minVersion)
+                        {
+                            unaffectedPages.Add(indexInfo.Items[i]);
+                        }
+                    }
+                }
+
+                // ACT
+                var result = await target.MergeAsync(indexInfo, sortedCatalog);
+
+                // ASSERT
+
+                // Verify the definitely unaffected pages are in still in the index and not loaded.
+                foreach (var page in unaffectedPages)
+                {
+                    Assert.Contains(page, indexInfo.Items);
+                    Assert.False(page.IsPageFetched, "A page before the lowest version input version must not be fetched.");
+                }
+
+                // Verify the modified version set.
+                Assert.Equal(modified, result.ModifiedLeaves.Select(x => x.Version).OrderBy(v => v).ToList());
+
+                // Verify the deleted version set.
+                Assert.Equal(deleted, result.DeletedLeaves.Select(x => x.Version).OrderBy(v => v).ToList());
+
+                // Verify the resulting set of versions.
+                var actualRemaining = await GetVersionsAsync(indexInfo);
+                Assert.Equal(expectedRemaining, actualRemaining);
+
+                // Verify all but the last page are full.
+                foreach (var pageInfo in indexInfo.Items.AsEnumerable().Reverse().Skip(1))
+                {
+                    Assert.True(
+                        pageInfo.Count == config.MaxLeavesPerPage,
+                        "All but the last page must have the maximum number of leaf items per page.");
+                }
+
+                // Verify the last page is not too full.
+                if (indexInfo.Items.Count > 0)
+                {
+                    Assert.True(
+                        indexInfo.Items.Last().Count <= config.MaxLeavesPerPage,
+                        "The last page must have less than or equal to the maximum number of leaf items per page.");
+                }
+
+                // Verify the page bounds are in ascending order.
+                var bounds = indexInfo.Items.SelectMany(GetUniqueBounds).ToList();
+                for (var i = 1; i < bounds.Count; i++)
+                {
+                    Assert.True(bounds[i - 1] < bounds[i], "Each page bound must be less than the next.");
+                }
+
+                // Verify the modified pages are in the index.
+                foreach (var pageInfo in result.ModifiedPages)
+                {
+                    Assert.True(indexInfo.Items.Contains(pageInfo), "A modified page must be in the index.");
+                }
+
+                // Verify the leaf item infos match the leaf items.
+                Assert.True(
```

Comment:
`Assert.Equal`

Reply:
During the debugging I found that the custom messages were more useful than the counts that come from `Assert.Equal`.